### PR TITLE
fix(eth/filters): tests

### PR DESCRIPTION
**Pull request by jonastheis**
This PR fixes the tests in `eth/filters`. All of the tests are RPC related.

```
# before changes of this PR
make test | tee test.out

# after changes of this PR
make test | tee test.out.zktrie

grep -e "--- FAIL:.\+ " -o test.out | sort > test.out.sorted
grep -e "--- FAIL:.\+ " -o test.out.zktrie  | sort > test.out.sorted.zktrie

diff test.out.sorted test.out.sorted.zktrie
9,10d8
< --- FAIL: TestClique 
< --- FAIL: TestClique/0 
24d21
< --- FAIL: TestEthClient/CallContract 
32d28
< --- FAIL: TestFilters 
55d50
< --- FAIL: TestHeaderVerification 
67,68c62
< --- FAIL: TestPendingTxFilterFullTx 
< --- FAIL: TestPrestateWithDiffModeTracer 
---
> --- FAIL: TestPrestateWithDiffModeTracer
77a72
> --- FAIL: TestShortRepair 
```

### CODE REVIEW

File: eth/filters/filter_test.go
Comment (0xmountaintop) [inline eth/filters/filter_test.go:1]: how do we know we should change to these hashes?

Comment (jonastheis) [reply eth/filters/filter_test.go:1]: Our genesis hash and state roots are different, therefore the hashes won't match the ones from upstream tests. I printed the block and tx hashes after the chain was generated and then adjusted the expected filter results accordingly.

Comment (0xmountaintop) [reply eth/filters/filter_test.go:1]: but why tx hashes also change?

Comment (0xmountaintop) [reply eth/filters/filter_test.go:1]: i c probably because of l2basefee/l1datafee

**Pull request merged**